rust/blockstore/src/arrow/ordered_blockfile_writer.rs

```diff
@@ -316,7 +316,8 @@ impl ArrowOrderedBlockfileWriter {
                 .current_block_delta
                 .take()
                 .expect("We already checked above that there is a current delta");
-            let new_delta = current_delta.split_off_half::<K, V>();
+
+            let new_delta = current_delta.split_off_last_key().expect("This returns None only if the delta is empty. We just added to the delta, so it is not empty.");
```

Comment:
This is the main change. Since the ordered writer writes keys in strictly increasing order, we can always split deltas at the last key instead of splitting them at their midpoint. This results in full blocks (rather than half-sized blocks) and better performance.